### ISSUE TITLE
[v2] Input text overflow

### ISSUE THREAD

**Issue by dbertella** [JedWatson/react-select#2774]
is this the expected behaviour for the input with a very long text?
![screen shot 2018-07-04 at 15 22 27](https://user-images.githubusercontent.com/2563196/42282442-6ce9ed68-7f9e-11e8-8f03-6b794bbbe78f.png)

To reproduce just go to the page with example and type a very long text

**Comment by lusa** [comment]
@dbertella I have a CSS hack to get around this, wrap your Select in a div and set CSS for the container's input to have a max-width. 
```
<div className="selectContainer">
    <Select ... />
</div> 

//CSS
.selectContainer input {
  max-width: 400px; //whatever width you need
}
```

**Comment by yanpolonskiy** [comment]
[https://vgy.me/cxBWTs.png](url)

> @dbertella I have a CSS hack to get around this, wrap your Select in a div and set CSS for the container's input to have a max-width. I tried adding this max-width in the input on styles props, but it did not work.

Cuz we have inline style change onInputing.

My hack is using !important for .Select-input >input, 
and setting width and max-width for .Select-input and for Select-multi-value-wrapper .
Also I set display: flex  for Select-multi-value-wrapper ( using display: table in 2k18, omfg)

**Comment by TheSharpieOne** [comment]
This works for me for v2 (including in IE 11) (note: "prefix" is the `classNamePrefix`)

```scss
.prefix__value-container {
  width: 90%;
}
.prefix__value-container--is-multi.prefix__value-container--has-value.prefix__value-container {
  width: 85%;
}
.prefix__value-container > div,
.prefix__input,
.prefix__input > input {
  max-width: 99%;
}
.prefix__input > input + div {
  max-width: 100%;
}
```

**Comment by jrhalchak** [comment]
Just to 👍 this even harder:

If your application uses CSS modules exclusively, hacking this to fix it is even more insane.

Had to resort to overriding style objects with flexbox and using nested CSS selectors like `div[style*='inline-block;']` to get it to work.

**Comment by jossmac** [comment]
Resolved by #2876
